src/uct/cuda/cuda_copy/cuda_copy_iface.c

```diff
@@ -133,22 +133,38 @@ static ucs_status_t uct_cuda_copy_iface_flush(uct_iface_h tl_iface, unsigned fla
 }
 
 static UCS_F_ALWAYS_INLINE unsigned
-uct_cuda_copy_progress_event_queue(uct_cuda_copy_iface_t *iface,
-                                   uct_cuda_copy_stream_t id,
-                                   unsigned max_events)
+uct_cuda_copy_queue_head_ready(ucs_queue_head_t *queue_head)
 {
-    unsigned count = 0;
     cudaError_t result = cudaSuccess;
     uct_cuda_copy_event_desc_t *cuda_event;
-    ucs_queue_iter_t iter;
 
-    ucs_queue_for_each_safe(cuda_event, iter,
-                            &iface->outstanding_event_q[id], queue) {
+    if (ucs_queue_is_empty(queue_head)) {
+        return 0;
+    } else {
+        cuda_event =
+            ucs_queue_head_elem_non_empty(queue_head, uct_cuda_copy_event_desc_t,
+                                          queue);
         result = cudaEventQuery(cuda_event->event);
         if (cudaSuccess != result) {
-            break;
+            return 0;
+        } else {
+            return 1;
```

Comment:
probably comon function is not a good idea
only this part is needed for ucs_queue_for_each_extract:
```
	    ucs_queue_for_each_extract(cuda_event, queue_head, queue,
                               cudaEventQuery(cuda_event->event) == cudaSuccess) {
```